lib/galaxy/auth/providers/ldap_ad.py

```diff
@@ -204,12 +209,18 @@ def authenticate(self, email, username, password, options):
             if not self._authenticate(params, options):
                 return failure_mode, '', ''
 
+        # check whether the user is a member of a specified group/domain/...
+        if 'search-memberof-filter' in options:
+            search_filter = _get_subs(options, 'search-memberof-filter', params)
+            if not any(search_filter in ad_node_name for ad_node_name in params['memberOf']):
+                return failure_mode, '', ''
+
         attributes = {}
         if self.auto_create_roles_or_groups:
             attributes['roles'] = params[self.role_search_option]
         return (True,
                 _get_subs(options, 'auto-register-email', params),
-                _get_subs(options, 'auto-register-username', params),
+                transform_publicname(_get_subs(options, 'auto-register-username', params), check_dup=False),
```

Comment:
See the other comment. Crash when called this way, without the `trans` parameter.  If there is no access to `trans ` in this context, maybe separate the public name transformation and the duplication check in two functions and only call the transformation specific one here ?

Reply:
Yes, sorry for the trouble, that's a good plan, I'll work on a patch.

Reply:
It should be fixed now.